src/rcc.rs

```diff
@@ -341,28 +371,36 @@ impl CFGR {
                 })
         });
 
-        Clocks {
+        let clocks = Clocks {
             hclk: Hertz(hclk),
             pclk1: Hertz(pclk1),
             pclk2: Hertz(pclk2),
             ppre1,
             ppre2,
             sysclk: Hertz(sysclk),
+            pll48clk,
+        };
+
+        if self.pll48clk {
+            assert!(clocks.is_pll48clk_valid());
         }
+
+        clocks
     }
 }
 
 /// Frozen clock frequencies
 ///
 /// The existence of this value indicates that the clock configuration can no longer be changed
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Debug)]
```

Comment:
Not a big fan of uneccessary `Debug` derives in embedded. They have the nasty tendency to massively blow up the generated code by accident.

Reply:
There is a dedicated commit on this ;-)

Reply:
Removed